app/controllers/activity_deletions_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class ActivityDeletionsController < BaseController
```

Comment:
I think I, in part, disagree with the group here in it's current implementation for a few reasons:
1. I'm not sure the actions make sense. An action is a user interaction, but they aren't creating a deletion.
2. It sends off the wrong request type `POST` instead of `DELETE`. I'm not sure if this necessarily breaks any rules, but it's definitely _less_ correct.
3. It goes against Rails conventions of having a `destroy` method for a resource.

I think point 2 and 3 could be fixed by making this a destroy, but then does ActivityDeletionsController#destroy make sense? Maybe? I'm not sure!

Reply:
Yeah this is an interesting one. I was initially thinking about just adding `confirm_destroy` and `destroy` to the `ActivityController`, but `confirm_destroy` doesn't feel very RESTful, which is why Ed suggested a deletions controller

I think you _could_ conceptualise the creation of an action, and in this case that action being deletion. It does seem a little strange given that `DELETE` is a method and destroy is a RESTful route, but we do have other controller where we're creating or updating _actions_ or doing other stuff that doesn't map neatly onto method/action + resource: `uploads#update` actually creating and updating activities; `redactions#update` actually updating an activity attribute; `forms#update` updating an activity

I agree it would be nice to use `DELETE` though, which could go back into the `ActivityController` as a `destroy` action, but I'm not sure what to do with the confirm deletion route unless we route `GET` and `DELETE` to separate actions or deal within it one action (but the latter feels like a single responsibility principle violation) 🤔

I suppose we have plenty of custom actions in other controllers, so `confirm_destroy` isn't too unconventional for RODA?

Reply:
Considering there are a few people who think this is the right direction then I'm sure this will be okay!

I do think `confirm_destroy` is what [most people land on](https://github.com/search?q=confirm_destroy+lang%3ARuby++path%3A%2F%5Eapp%5C%2Fcontrollers%5C%2F%2F&type=code)

Reply:
Did something go wrong with that link? 🤔

<img width="656" alt="image" src="https://user-images.githubusercontent.com/40244233/227512650-d3bc7653-bb74-48cc-ab3a-dd7dda544dad.png">

Reply:
Works for me https://github.com/search?q=confirm_destroy+lang%3ARuby++path%3A%2F%5Eapp%5C%2Fcontrollers%5C%2F%2F&type=code 
<img width="467" alt="image" src="https://user-images.githubusercontent.com/47089130/227515570-ed7063f1-5aba-47ca-bb2c-68d83ecf48f1.png">

Reply:
Bizarre - I get nothing 🤔

Reply:
Ah, this works! Looks like it's pretty common then. Whether it's right or right-er is another matter 😅 I'm still open to persuasion

Reply:
It was good enough for RailsCasts in 2007: http://railscasts.com/episodes/77-destroy-without-javascript

Reply:
I suppose the fact that it's hard to decide is proof enough that there's no "one right way".

Some questions I'd ask myself are things like: is one or the other choice more likely to be hit by future changes in Rails (for example if we were relying on an HTTP method being used in a way that might not be supported in future), are they equally maintainable and understandable for future teams etc.

Reply:
Maybe better to stick with `destroy` and `DELETE` for destroying in that case!